src/test/java/javaslang/collection/AbstractMapTest.java

```diff
@@ -99,6 +99,11 @@
         return IntMap.of(emptyMap());
     }
 
+    @Override
+    protected boolean emptyShouldBeSingleton() {
+        return false;
```

Comment:
For non-sorted Maps, e.g. HashMap, it is a singleton (I think).
For sorted Maps, e.g. TreeMap, it is no singleton because of the Comparator.
Maybe we can add a `abstract class AbstractSortedMapTest extends AbstractMapTest` to overwrite it for sorted Maps?